sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MethodShouldBeNameAccordingToSynchronicity.cs

```diff
@@ -69,7 +69,8 @@ protected override void Initialize(SonarAnalysisContext context)
                 c =>
                 {
                     var methodDeclaration = (MethodDeclarationSyntax)c.Node;
-                    if (methodDeclaration.Identifier.IsMissing)
+                    if (methodDeclaration.Identifier.IsMissing ||
+                        methodDeclaration.Identifier.ValueText == "Main")

```

Comment:
I could have done a more strict check to ensure this is a real main method but I think that's good enough like this.

Reply:
Why not call _SonarAnalyzer.Helpers.KnownMethods.IsMainMethod_?
It uses the symbol, but the code fetches that in line 78 anyway so it isn't going to affect perf much.

Reply:
Well except that for all methods named `Main` you will actually not call the semantic model so you are faster.